### PR TITLE
Create automated test for the GitHub PR plugin

### PR DESCRIPTION
**Pull request by Ohrimenko1988**
### What does this PR do?
Create automated test for the GitHub PR plugin

Issue: https://github.com/eclipse/che/issues/19513

### PR Checklist

[As the author of this Pull Request I made sure that:](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#pull-request-template-and-its-checklist)

- [ ] [The Eclipse Contributor Agreement is valid](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-eclipse-contributor-agreement-is-valid)
- [ ] [Code produced is complete](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-produced-is-complete)
- [ ] [Code builds without errors](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#code-builds-without-errors)
- [ ] [Tests are covering the bugfix](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#tests-are-covering-the-bugfix)
- [ ] [The repository devfile is up to date and works](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#the-repository-devfile-is-up-to-date-and-works)
- [ ] [Sections `What issues does this PR fix or reference` and `How to test this PR` completed](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#sections-what-issues-does-this-pr-fix-or-reference-and-how-to-test-this-pr-completed)
- [ ] [Relevant user documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [Relevant contributing documentation updated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#relevant-contributing-documentation-updated)
- [ ] [CI/CD changes implemented, documented and communicated](https://github.com/eclipse/che/blob/master/CONTRIBUTING.md#cicd-changes-implemented-documented-and-communicated)

### Reviewers

Reviewers, please comment how you tested the PR when approving it.

### CODE REVIEW

File: tests/e2e/TestConstants.ts [hunk 1]
```diff
@@ -17,12 +17,26 @@ function getBaseUrl(): string {
     return baseUrl.replace(/\/$/, '');
 }
 
+function getKeycloackUrl(): string {
+    const keycloakUrl: string | undefined = process.env.TS_SELENIUM_KEYCLOAK_URL;
+    if (!keycloakUrl) {
+        return 'http://keycloak-url';
+    }
+
+    return keycloakUrl.replace(/\/$/, '');
```
Comment (ScrewTSW) [inline tests/e2e/TestConstants.ts]: What is this for? I don't understand the meaning of this regex

Comment (Ohrimenko1988) [reply tests/e2e/TestConstants.ts]: The same as for the "TS_SELENIUM_BASE_URL". Removal of the "/" character at the end of the introduced URL if that url has been set with that character.

**Comment by Ohrimenko1988** [comment]
[crw-ci-test]

**Review by ScrewTSW — APPROVED** [review]

File: tests/e2e/TestConstants.ts [hunk 2]
```diff
@@ -17,12 +17,26 @@ function getBaseUrl(): string {
...
 export const TestConstants = {
     /**
      * Base URL of the application which should be checked
      */
     TS_SELENIUM_BASE_URL: getBaseUrl(),
 
+    /**
+     * The keycloak URL of the application which should be checked
+     */
+    TS_SELENIUM_KEYCLOAK_URL: getKeycloackUrl(),
```
Comment (Katka92) [inline tests/e2e/TestConstants.ts]: I'm a little bit confused - so this is not the Keycloak URL of Che, but the Keycloak URL of the running application?

Comment (Ohrimenko1988) [reply tests/e2e/TestConstants.ts]: Fixed

File: tests/e2e/utils/KeycloackUrlHandler.ts [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+
+import { TestConstants } from '../TestConstants';
+
+export class KeycloackUrlHandler {
+    public static getBaseKeycloakUrl(): string {
+        let baseKeycloakUrl: string = TestConstants.TS_SELENIUM_BASE_URL;
+
+        if (!TestConstants.TS_SELENIUM_SINGLE_HOST) {
+            baseKeycloakUrl = baseKeycloakUrl.replace('che', 'keycloak');
+        }
```
Comment (ScrewTSW) [inline tests/e2e/utils/KeycloackUrlHandler.ts:19]: no single-host url being returned

File: tests/e2e/utils/requestHandlers/tokens/CheMultiuserTokenHandler.ts [hunk 4]
```diff
@@ -12,22 +12,14 @@ import querystring from 'querystring';
...
-        let params = {};
+        const keycloakUrl = KeycloackUrlHandler.getTokenEndpointUrl();
 
-        let keycloakUrl = this.handleTrailingSlash(TestConstants.TS_SELENIUM_BASE_URL);
-        const keycloakAuthSuffix = 'auth/realms/che/protocol/openid-connect/token';
-
-        if (TestConstants.TS_SELENIUM_SINGLE_HOST) {
-            keycloakUrl = keycloakUrl + keycloakAuthSuffix;
-        } else {
-            keycloakUrl = keycloakUrl.replace('che', 'keycloak') + keycloakAuthSuffix;
```
Comment (ScrewTSW) [inline tests/e2e/utils/requestHandlers/tokens/CheMultiuserTokenHandler.ts:27]: This case has been completely omitted with the removal

**Review by ScrewTSW — CHANGES_REQUESTED** [review]

File: tests/e2e/utils/KeycloackUrlHandler.ts (hunk 3 above)
Comment (Ohrimenko1988) [reply tests/e2e/utils/KeycloackUrlHandler.ts:19]: I don't understand what do you mean.
If the "TS_SELENIUM_SINGLE_HOST" is false it returns multihost url
If the "TS_SELENIUM_SINGLE_HOST" is true returns single-host url

examples from logs:
 -  https://che-iokhrime-che.192.168.99.254.nip.io/auth/realms/che/protocol/openid-connect/token
 -  https://keycloak-iokhrime-che.192.168.99.254.nip.io/auth/realms/che/protocol/openid-connect/token

File: tests/e2e/utils/requestHandlers/tokens/CheMultiuserTokenHandler.ts (hunk 4 above)
Comment (Ohrimenko1988) [reply tests/e2e/utils/requestHandlers/tokens/CheMultiuserTokenHandler.ts:27]: No, it is handling in the "getBaseKeycloakUrl".
I would say that it was completely simplified.

File: tests/e2e/utils/KeycloackUrlHandler.ts (hunk 3 above)
Comment (Ohrimenko1988) [reply tests/e2e/utils/KeycloackUrlHandler.ts:19]: I have checked it once more:
 - Single-host: https://che-iokhrime-che.192.168.99.254.nip.io/auth/realms/che/protocol/openid-connect/token
 - Multi-host  : https://keycloak-iokhrime-che.192.168.99.254.nip.io/auth/realms/che/protocol/openid-connect/token

Comment (ScrewTSW) [reply tests/e2e/utils/KeycloackUrlHandler.ts:19]: sorry, my bad. I misread the code.

File: tests/e2e/utils/requestHandlers/tokens/CheMultiuserTokenHandler.ts (hunk 4 above)
Comment (ScrewTSW) [reply tests/e2e/utils/requestHandlers/tokens/CheMultiuserTokenHandler.ts:27]: I see it now, sorry

**Review by ScrewTSW — APPROVED** [review]

**Pull request merged**